UI/SidePanel.cpp

```diff
@@ -2041,85 +2034,70 @@ void SidePanel::PlanetPanel::RClick(const GG::Pt& pt, GG::Flags<GG::ModKey> mod_
         return;
     }
 
-    GG::MenuItem menu_contents;
+    CUIPopupMenu popup(pt.x, pt.y);
+
     if (planet->OwnedBy(HumanClientApp::GetApp()->EmpireID()) && m_order_issuing_enabled
         && m_planet_name->InClient(pt))
     {
-        menu_contents.next_level.push_back(GG::MenuItem(UserString("SP_RENAME_PLANET"), 1, false, false));
+        auto rename_action = [this, planet]() { // rename planet
+            std::string plt_name = planet->Name();
+            CUIEditWnd edit_wnd(GG::X(350), UserString("SP_ENTER_NEW_PLANET_NAME"), plt_name);
+            edit_wnd.Run();
+            if (edit_wnd.Result() != "" && edit_wnd.Result() != planet->Name() && m_order_issuing_enabled)
```

Comment:
possibly also break this over 2+ lines, especially since the { is on a separate line anyway